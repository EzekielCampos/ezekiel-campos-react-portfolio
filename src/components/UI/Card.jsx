import {
  Grid2 as Grid,
  Card,
  CardActions,
  CardContent,
  CardMedia,
  Typography,
  IconButton,
} from "@mui/material";
import GitHubIcon from "@mui/icons-material/GitHub";
import LanguageIcon from "@mui/icons-material/Language"; // For deployed page icon
import PropTypes from "prop-types";

export const ProjectCard = ({ data }) => {
  console.log(data);

  return data.map((data) => {
    return (
      <Grid size={{ xs: 12, md: 6, s: 4 }} key={data.id}>
        <Card sx={{ display: "flex", flexDirection: "column", height: "100%" }}>
          <CardMedia
            sx={{ height: 400, objectFit: "cover" }}
            image={data.pic}
            title={data.name}
          />

          {/* Card Content */}
          <CardContent>
            <Grid container justifyContent={"center"}>
              <Typography gutterBottom variant="h5" component="div">
                {data.name}
              </Typography>
            </Grid>
            <Grid container justifyContent={"center"}>
              <Typography variant="body2" sx={{ color: "text.secondary" }}>
                {data.description}
              </Typography>
            </Grid>
          </CardContent>

          {/* Card Actions */}
          <CardActions sx={{ justifyContent: "center", marginTop: "auto" }}>
            {/* GitHub Icon */}
            <IconButton
              sx={{ color: "skyBlue" }}
              onClick={() => window.open(data.github)}
            >
              <GitHubIcon style={{ width: 35, height: 35 }} />
            </IconButton>

            {/* Deployed Page Icon */}
            <IconButton
              sx={{ color: "#000" }}
              //   onClick={() => window.open(deployedUrl, "_blank")}
            >
              <LanguageIcon style={{ width: 35, height: 35 }} />
            </IconButton>
          </CardActions>
        </Card>
      </Grid>
    );
  });
};

ProjectCard.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string, // `title` should be a string
      description: PropTypes.string, // `description` should be a string
      github: PropTypes.string,
      pic: PropTypes.string,
    })
  ).isRequired,
};
