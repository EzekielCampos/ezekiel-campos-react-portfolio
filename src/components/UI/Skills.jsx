import {
    ListItem,
    ListItemIcon,
    ListItemText,
  } from "@mui/material";
  


  export const Skills = ({data})=>{

    return(data.map((item)=>{
        return(
            <ListItem key={item.id}>
            <ListItemIcon style={{ minWidth: "auto", marginRight: "8px" }}>
              •
            </ListItemIcon>
            <ListItemText>
              {item.tech}
            </ListItemText>
          </ListItem>

        )
    })
       
    )

  }



{/* <ListItem>
<ListItemIcon style={{ minWidth: "auto", marginRight: "8px" }}>
  •
</ListItemIcon>
<ListItemText>
  First Item
  <GitHubIcon
    style={{ width: 20, height: 20, paddingLeft: "5px" }}
  />
</ListItemText>
</ListItem>
<ListItem>
<ListItemIcon style={{ minWidth: "auto", marginRight: "8px" }}>
  •
</ListItemIcon>
<ListItemText fontSize={10}>
  second Item
  {/* <ListItemIcon sx={{ width: 20, height: 20, color:"black" }}>{reactIcon}</ListItemIcon> */}
  {/* React icon */}
// </ListItemText>
// </ListItem>
// <ListItem>
// <ListItemIcon style={{ minWidth: "auto", marginRight: "8px" }}>
//   •
// </ListItemIcon>
// <ListItemText>
//   second Item
//   <JavascriptIcon sx={{ width: 20, height: 20 }} />{" "}
// </ListItemText>
// </ListItem>

// <ListItem>
// <ListItemIcon style={{ minWidth: "auto", marginRight: "8px" }}>
//   •
// </ListItemIcon>
// <ListItemText>
//   second Item
//   <ListItemIcon>{mongoIcon}</ListItemIcon>
// </ListItemText>
// </ListItem> */}