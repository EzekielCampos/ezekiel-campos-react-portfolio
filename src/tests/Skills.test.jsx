import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { Skills } from "../components/UI/Skills"
import pretty from "pretty";


export const skill = [
  {
    id: 1,
    tech: "JavaScript",
  },];

describe("Skills Item", () => {
  it("renders the skills item component and matches snapshot", () => {
    // Render the component
    render(<Skills data = {skill}/>);

    // Format the rendered output
    const html = pretty(document.querySelector(".skill").innerHTML);

    // Verify the output
    expect(html).toMatchSnapshot();
  });
});
