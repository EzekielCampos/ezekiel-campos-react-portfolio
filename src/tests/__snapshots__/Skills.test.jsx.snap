// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > renders the Nav component and matches snapshot 1`] = `
"<div class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root" style="min-width: auto; margin-right: 8px;">•</div>
<div class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"><span class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root">JavaScript</span></div>"
`;

exports[`Skills Item > renders the  component and matches snapshot 1`] = `
"<div class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root" style="min-width: auto; margin-right: 8px;">•</div>
<div class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"><span class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root">JavaScript</span></div>"
`;

exports[`Skills Item > renders the Nav component and matches snapshot 1`] = `
"<div class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root" style="min-width: auto; margin-right: 8px;">•</div>
<div class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"><span class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root">JavaScript</span></div>"
`;

exports[`Skills Item > renders the skills item component and matches snapshot 1`] = `
"<div class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root" style="min-width: auto; margin-right: 8px;">•</div>
<div class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"><span class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root">JavaScript</span></div>"
`;

exports[`renders the Nav component and matches snapshot 1`] = `
"<div class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root" style="min-width: auto; margin-right: 8px;">•</div>
<div class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"><span class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root">JavaScript</span></div>"
`;
